### PR TITLE
assume::posint #144 fix

### PR DESCRIPTION
**Pull request by aaditya-thakkar**

### CODE REVIEW

**Comment by aaditya-thakkar** [comment]
Is it okay or something more needs to be done?

File: ginac/symbol.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ void symbol::do_print_python_repr(const print_python_repr & c, unsigned level) c
 void symbol::set_domain(unsigned d)
 {
         domain = d;
-        iflags.clear();
+        //iflags.clear();
```
Comment (rwst) [inline ginac/symbol.cpp]: In future please simply delete and leave no outcommented code. It only adds to any confusion.

**Comment by rwst** [comment]
Please always test your branch or patch with at least the relevant Sage dotests, you cannot rely on me always finding the errors. Errors are inevitable, but you must find and catch thm yourself. With Pynac it suffices in the first round to doctest `src/sage/symbolic`, `src/sage/functions`, `src/sage/calculus`, and `src/doc`. You will find errors in ``symbolic/assumptions.py`.

**Comment by aaditya-thakkar** [comment]
Sorry, it won't happen again.  With this I have modified my code but it still fails 3 tests, which I guess  are related with function expansions and simplifications. Here is the screenshot attachedby. Before the changes these tests were failing too. I have checked manually for those 3 cases that since `x` is assumed to be integer, it gives `True` before calling `sin(x*pi)` or `sin(x*pi).simplify()` , but according to doctests `sin(x*pi)` should return `sin(x*pi)` and `sin(x*pi).simplify()` should return ` 0`, my patch gives `0 ` for both.  
screenshot from 2016-03-03 18-34-38

**Comment by behackl** [comment]
The failures are caused by #132, they are fixed with the pynac upgrade in [trac #20134](http://trac.sagemath.org/ticket/20134).

**Comment by rwst** [comment]
There is a change in procedure, from now, because of the number of developers. Doctests may fail in master before the freezing phase of a new release. During freeze they must be all fixed. The second change in procedure is that for every PR that needs changes in Sage to pass tests (both Python/Cython code fixes and doctest adaptations) those changes must be attached to the respective PR, so that I can collect them for the Sage ticket.

I will put this also on a wiki page for everyone to see.

**Comment by aaditya-thakkar** [comment]
For now, should I send the pull request with the modifications?

**Comment by rwst** [comment]
The Sage modifications should be "sent" as PR comment. Just make diff, copy and paste into a code block.

Let me add that of course every PR must compile without warnings before it gets accepted.

**Comment by aaditya-thakkar** [comment]
`-        //iflags.clear();
         switch (d) {
+                
                 case domain::complex:
+                        iflags.set(info_flags::real, false);
+                        iflags.set(info_flags::positive, false);
+                        iflags.set(info_flags::nonnegative, false);
+                        iflags.set(info_flags::integer, false);
                         break;
                 case domain::real:
                         iflags.set(info_flags::real, true);
+                        if(iflags.to_string()=="positive") {
+                                iflags.set(info_flags::positive, true);
+                                iflags.set(info_flags::nonnegative, true);
+                        }  
+                        else {
+                                iflags.set(info_flags::positive, false);
+                                iflags.set(info_flags::nonnegative, false);
+                        }
+                        if(iflags.to_string()=="integer") {
+                                iflags.set(info_flags::integer, true);
+                        }
+                        else {
+                                iflags.set(info_flags::integer, false);
+                        }
                         break;
                 case domain::positive:
                         iflags.set(info_flags::real, true);
@@ -187,6 +205,11 @@ void symbol::set_domain(unsigned d)
                 case domain::integer:
                         if(iflags.to_string()=="positive") {
                                 iflags.set(info_flags::positive, true);
+                                iflags.set(info_flags::nonnegative, true);
+                        }
+                        else {
+                                iflags.set(info_flags::positive, false);
+                                iflags.set(info_flags::nonnegative, false);
                         }
                         iflags.set(info_flags::real, true);
                         iflags.set(info_flags::integer, true);
(END)
`

**Comment by behackl** [comment]
@rwst This is an excellent idea. What would be the policy w.r.t. tickets like

- http://trac.sagemath.org/ticket/20102
- http://trac.sagemath.org/ticket/20099
- http://trac.sagemath.org/ticket/20098

etc.? I.e. do you still want to create single "doctest"-tickets for these failures which have to be provided after the actual pynac release, or would you add these things on the update-ticket as well?

@aaditya-thakkar I don't think you have to provide anything in this particular case, because this **is** already fixed.

**Comment by aaditya-thakkar** [comment]
@behackl  I didn't get it. Is this issue solved already?

**Comment by behackl** [comment]
> @behackl I didn't get it. Is this issue solved already?

Yes, Ralf has already fixed those three failing doctests that you found with this branch in the branch attached [to this trac ticket](http://trac.sagemath.org/ticket/20134). At least as far I can tell.

**Comment by rwst** [comment]
@aaditya-thakkar What I meant is the Sage changes not the Pynac diff.

https://github.com/pynac/pynac/wiki/Development-protocol

**Comment by rwst** [comment]
I see, I'll add that please keep your local master up-to-date.

**Comment by rwst** [comment]
Maybe we need a repo for the Sage changes since the last Pynac update in Sage.

**Comment by rwst** [comment]
@behackl No the tickets where Sage problems are reported that turn out to be caused in Pynac, they stay as occasion to add respective doctests. What I meant was changes needed to make Pynac pass a `make ptestlong` with Sage.

**Comment by behackl** [comment]
What about creating a new "pynac upgrade" trac ticket with a public branch to which we can push our doctest changes on our own?

The ticket would be set to `needs_review` if a new pynac version gets released.

**Comment by behackl** [comment]
> @behackl No the tickets where Sage problems are reported that turn out to be caused in Pynac, they stay as occasion to add respective doctests. What I meant was changes needed to make Pynac pass a `make ptestlong` with Sage.

Of course, we can't prohibit the creation of these tickets; that's not really what I meant: sometimes I created a trac ticket after reporting the issue here. And that is probably what we/I would not need to do any longer.

**Comment by rwst** [comment]
There is now a new site https://github.com/pynac/sage that will have branches for people working with development versions of Pynac. I have extended the devbelopment protocol wiki page with all suggestions, in particular @behackl the Sage doctests can be either text or branch, as Pynac PR comment, Pynac issue comment, Sage branch, or Sage ticket comment.

**Comment by rwst** [comment]
See for example https://github.com/pynac/sage/tree/for-pynac-0.6.3
You can see from the commit message of which fails I'm aware. You can of course also do PRs for that branch.

**Comment by rwst** [comment]
Apologies, that should have been named "for-pynac-0.6.4".

**Comment by behackl** [comment]
Alright, I understand. I've updated the wiki entry to clarify the process.

**Comment by rwst** [comment]
Please use https://groups.google.com/forum/?hl=en#!forum/pynac-devel for general discussion from now on.

**Comment by aaditya-thakkar** [comment]
As discussed above, it is passing all the tests except those 3 that were mentioned. Please review it and suggest if something is missing.

**Comment by rwst** [comment]
You have added whitespace, please remove. All else is fine AFAI can see.

**Comment by aaditya-thakkar** [comment]
Removed! :)

File: ginac/symbol.cpp [hunk 2]
```diff
@@ -171,19 +171,45 @@ void symbol::do_print_python_repr(const print_python_repr & c, unsigned level) c
...
+                        }
+                        else {
+                                iflags.set(info_flags::integer, false);
+                        }
                         break;
                 case domain::positive:
                         iflags.set(info_flags::real, true);
                         iflags.set(info_flags::positive, true);
                         iflags.set(info_flags::nonnegative, true);
+                        iflags.set(info_flags::integer, true);
```
Comment (rwst) [inline ginac/symbol.cpp]: That slipped the radar.

**Pull request merged**